splice_machine/src/main/java/com/splicemachine/derby/utils/SpliceAdmin.java

```diff
@@ -1283,6 +1283,9 @@ public static void SYSCS_SET_GLOBAL_DATABASE_PROPERTY(final String key,final Str
             PropertyInfo.setDatabaseProperty(key, value);
             DDLMessage.DDLChange ddlChange = ProtoUtil.createSetDatabaseProperty(tc.getActiveStateTxn().getTxnId(), key);
             tc.prepareDataDictionaryChange(DDLUtils.notifyMetadataChange(ddlChange));
+            // we need to invalidate the statement caches since we could set parameters that affect query plans.
+            SYSCS_INVALIDATE_STORED_STATEMENTS();
+            SYSCS_EMPTY_GLOBAL_STATEMENT_CACHE();
```

Comment:
I like the idea of automatically forcing recompile for convenience, but are you sure we want to do this?  It may have a performance impact.  Every time a flag is changed we have to recompile all SQLs and triggers.  There is no option for the user to set a database property without calling these cache-clearing functions if they wish to avoid it.  Something we should think about, perhaps, before making this change.

Reply:
We can decide this point later, so the build can proceed.  Opened the following Jira:  https://splicemachine.atlassian.net/browse/DB-10948

Reply:
Thanks @msirek for your observation, maybe it is OK to keep this implementation as setting a global database property doesn't happen very often but let's follow up in the JIRA you opened to discuss improvements (one improvement that @dgomezferro mentioned was e.g. to only invalidated affected statements using the dependency manager).

Reply:
This is not needed for all database properties. You can do this if the property is timestamp precision

Reply:
Thanks @jyuanca! I will address it in DB-10948 then.